e2e/interchaintestv8/e2esuite/suite.go

```diff
@@ -2,64 +2,101 @@ package e2esuite
 
 import (
 	"context"
+	"os"
 
 	dockerclient "github.com/docker/docker/client"
 	"github.com/stretchr/testify/suite"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zaptest"
 
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+
 	sdkmath "cosmossdk.io/math"
 
 	interchaintest "github.com/strangelove-ventures/interchaintest/v8"
 	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
-	"github.com/strangelove-ventures/interchaintest/v8/chain/ethereum"
+	icethereum "github.com/strangelove-ventures/interchaintest/v8/chain/ethereum"
 	"github.com/strangelove-ventures/interchaintest/v8/ibc"
 	"github.com/strangelove-ventures/interchaintest/v8/testreporter"
 
 	"github.com/srdtrk/solidity-ibc-eureka/e2e/v8/chainconfig"
+	"github.com/srdtrk/solidity-ibc-eureka/e2e/v8/ethereum"
 	"github.com/srdtrk/solidity-ibc-eureka/e2e/v8/testvalues"
 )
 
+const (
+	TestnetTypePoW = "pow"
+	TestnetTypePoS = "pos"
+)
+
 // TestSuite is a suite of tests that require two chains and a relayer
 type TestSuite struct {
 	suite.Suite
 
-	ChainA       *ethereum.EthereumChain
-	ChainB       *cosmos.CosmosChain
-	UserA        ibc.Wallet
-	UserB        ibc.Wallet
-	dockerClient *dockerclient.Client
-	network      string
-	logger       *zap.Logger
-	ExecRep      *testreporter.RelayerExecReporter
+	ChainA         ethereum.Ethereum
+	ethTestnetType string
+	ChainB         *cosmos.CosmosChain
+	UserB          ibc.Wallet
+	dockerClient   *dockerclient.Client
+	network        string
+	logger         *zap.Logger
+	ExecRep        *testreporter.RelayerExecReporter
+
+	EthereumLightClientID         string
+	TendermintLightClientID       string
+	LastEtheruemLightClientUpdate uint64
+
+	// proposalIDs keeps track of the active proposal ID for cosmos chains
+	proposalIDs map[string]uint64
 }
 
 // SetupSuite sets up the chains, relayer, user accounts, clients, and connections
 func (s *TestSuite) SetupSuite(ctx context.Context) {
-	chainSpecs := chainconfig.DefaultChainSpecs
+	t := s.T()
```

Comment:
I don't usually like this. Why not just use `s.T()` each time instead of `t`

Reply:
Changed